README.md

```diff
@@ -50,116 +50,83 @@ Stay up to date with the latest ONNX news. [[Facebook](https://www.facebook.com/
-
-```
-sudo apt-get install protobuf-compiler libprotoc-dev
-pip install onnx
+# prefer lite proto
+set CMAKE_ARGS=-DONNX_USE_LITE_PROTO=ON
+pip install -e .
 ```
 
-### Windows
-If you are building ONNX from source on Windows, it is recommended that you also build Protobuf locally as a static library. The version distributed with conda-forge is a DLL, but ONNX expects it to be a static library.
-
-Note that the instructions in this README assume you are using Visual Studio. It is recommended that you run all the commands from a shell started from "Developer Command Prompt for VS 2019" and keep the build system generator for cmake (e.g., cmake -G "Visual Studio 16 2019") consistent.
-
-#### Build Protobuf and ONNX on Windows
-Step 1: Build Protobuf locally
+### Building Protobuf from source
+* **Windows**
 ```
 git clone https://github.com/protocolbuffers/protobuf.git
 cd protobuf
 git checkout v3.11.3
 cd cmake
-# Explicitly set -Dprotobuf_MSVC_STATIC_RUNTIME=OFF to make sure protobuf does not statically link to runtime library
-cmake -G -A -Dprotobuf_MSVC_STATIC_RUNTIME=OFF -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_BUILD_EXAMPLES=OFF -DCMAKE_INSTALL_PREFIX=<protobuf_install_dir>
-# For example:
-# cmake -G "Visual Studio 16 2019" -A x64 -Dprotobuf_MSVC_STATIC_RUNTIME=OFF -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_BUILD_EXAMPLES=OFF -DCMAKE_INSTALL_PREFIX=..\install
+cmake -G "Visual Studio 16 2019" -A $arch -DCMAKE_INSTALL_PREFIX=<protobug_install_dir> -DCMAKE_BUILD_TYPE=Release -Dprotobuf_MSVC_STATIC_RUNTIME=ON -DProtobuf_USE_STATIC_LIBS=ON -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_BUILD_EXAMPLES=OFF .
 msbuild protobuf.sln /m /p:Configuration=Release
 msbuild INSTALL.vcxproj /p:Configuration=Release
 ```
 
-Step 2: Build ONNX
+* **Linux**
 ```
-# Get ONNX
-git clone https://github.com/onnx/onnx.git
-cd onnx
+git clone https://github.com/protocolbuffers/protobuf.git
+cd protobuf
+git checkout v3.11.3
 git submodule update --init --recursive
-
-# Set environment variables to find protobuf and turn off static linking of ONNX to runtime library.
-# Even better option is to add it to user\system PATH so this step can be performed only once.
-# For more details check https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=vs-2017
-set PATH=<protobuf_install_dir>\bin;<protobuf_install_dir>\include;<protobuf_install_dir>\libs;%PATH%
-set USE_MSVC_STATIC_RUNTIME=0
-
-# use the static installed protobuf
-set CMAKE_ARGS=-DONNX_USE_PROTOBUF_SHARED_LIBS=OFF -DProtobuf_USE_STATIC_LIBS=ON
-
-# Optional: Set environment variable `ONNX_ML=1` for onnx-ml
-
-# Build ONNX
-python setup.py install
+mkdir build_source && cd build_source
+cmake ../cmake -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_POSITION_INDEPENDENT_CODE=ON -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Release
```

Comment:
protobuf_BUILD_SHARED_LIBS is a better option than BUILD_SHARED_LIBS since BUILD_SHARED_LIBS can be overwritten by passing a different value for protobuf_BUILD_SHARED_LIBS. It also makes it consistent with Windows.